File: 77. Combinations/77. Combinations.cs
public class Solution {
    public IList<IList<int>> Combine(int n, int k) {
        var result = new List<IList<int>>();
        Backtrack(n, k, 1, new List<int>(), result);
        return result;
    }
    
    private void Backtrack(int n, int k, int start, List<int> current, List<IList<int>> result) {
        if (current.Count == k) {
            result.Add(new List<int>(current));
            return;
        }
        
        for (int i = start; i <= n; i++) {
            current.Add(i);
            Backtrack(n, k, i + 1, current, result);
            current.RemoveAt(current.Count - 1);
        }
    }
}

File: 77. Combinations/77. Combinations.dart
class Solution {
  List<List<int>> combine(int n, int k) {
    List<List<int>> result = [];
    backtrack(n, k, 1, [], result);
    return result;
  }
  
  void backtrack(int n, int k, int start, List<int> current, List<List<int>> result) {
    if (current.length == k) {
      result.add(List.from(current));
      return;
    }
    
    for (int i = start; i <= n; i++) {
      current.add(i);
      backtrack(n, k, i + 1, current, result);
      current.removeLast();
    }
  }
}